solc/CommandLineInterface.cpp

```diff
@@ -416,8 +418,10 @@ void CommandLineInterface::readInputFilesAndConfigureRemappings()
 				auto infile = boost::filesystem::path(path);
 				if (!boost::filesystem::exists(infile))
 				{
-					cerr << "Skipping non-existent input file \"" << infile << "\"" << endl;
-					continue;
+					if(! m_args.count(g_argIgnoreMissingFiles)){
+						cerr << "Non-existent input file " << infile << endl;
+						exit(1);
+					}
 				}
 
 				if (!boost::filesystem::is_regular_file(infile))
```

Comment:
I think this should be considered part of this. It just does skipping if "file is invalid".

Reply:
@axic Ah, I can update that section as well.  To be clear, if the file is missing OR invalid I should communicate to the user and error out.  That's what you are requesting.

Reply:
@chriseth are you also ok with that?

Reply:
yes

Reply:
@axic @chriseth Alright, officially added as scope.